src/components/uploadPicture/uploadImage.tsx

```diff
@@ -0,0 +1,190 @@
+    setShowStart(true);
+    setStartButtonDisabled(true);
+    setUploading(true);
+  };
+
+  const getPresignedUrlForUpload =
+    trpc.file.getPresignedUrlForFileUpload.useMutation();
+  const finishFileUpload = trpc.file.finishFileUpload.useMutation();
+
+  const uploadImageToCloud = useCallback(async () => {
+    if (!capturedImage) return;
+
+    const { key, url } = await getPresignedUrlForUpload.mutateAsync({
+      type: capturedImage.type || "application/octet-stream",
+    });
+
+    await axios.put(url, capturedImage);
+
+    await finishFileUpload.mutateAsync({
+      key,
+      filename: fileName,
+    });
+
+    setUploadSuccess(true);
+
+    onUpload(key);
+
+    setShowStart(true);
+    setStartButtonDisabled(false);
+    setUploading(false);
+  }, [
+    capturedImage,
+    fileName,
+    getPresignedUrlForUpload,
+    finishFileUpload,
+    onUpload,
+  ]);
+
+  useEffect(() => {
+    if (uploading) {
+      void uploadImageToCloud();
+    }
+  }, [uploading, uploadImageToCloud]);
+
+  useEffect(() => {
+    if (uploadSuccess) {
+      const timeout = setTimeout(() => {
+        setUploadSuccess(false);
+      }, 2000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [uploadSuccess]);
+
+  return (
+    <div>
+      {showStart && (
+        <button
+          className={
+            startButtonDisabled
+              ? styles.startButtonDisabled
+              : styles.startButton
+          }
+          onClick={onClickStartButton}
+          disabled={startButtonDisabled}
+        >
+          <div className={styles.startButtonContent}>
+            <CameraIcon />
+            <span>{title}</span>
+          </div>
+        </button>
+      )}
+      {showTakePicture && (
+        <>
+          <input
+            type="file"
+            accept="image/*;capture=camera"
+            capture="environment"
+            onChange={handleFile}
+            className={styles.startButton}
+          />
+          <button className={styles.startButton} onClick={onCancelTakePicture}>
+            Cancel
+          </button>
+        </>
+      )}
+      {showUploadPicture && (
+        <div className={styles.uploadedImageContainer}>
+          {capturedImage && (
+            <svg>
+              <image
+                href={URL.createObjectURL(capturedImage)}
+                // alt="image"
+                className={styles.uploadedImage}
+              />
+            </svg>
+          )}
+          <div className={styles.buttonsContainer}>
+            <button
+              className={styles.startButton}
+              onClick={onClickUploadPicture}
+            >
+              Upload
+            </button>
+            <button
+              className={styles.startButton}
+              id="cancel-button"
+              onClick={onCancelCaptureImage}
+            >
+              Cancel
+            </button>
+          </div>
+        </div>
+      )}
+
+      {uploading && <p id="uploading">Uploading.......</p>}
```

Comment:
🐧 MUI comes with [loading buttons](https://mui.com/material-ui/react-button/#loading-button) that we could use here